### PR TITLE
Fix scala 2.13 warnings and update sbt

### PR DESCRIPTION
**Pull request by ekiwi**

### CODE REVIEW

**Comment by ekiwi** [comment]
No backport since this changes the function signature for `join`

File: src/main/scala/chiseltest/internal/BackendInterface.scala [hunk 1]
```diff
@@ -11,13 +11,13 @@ import scala.collection.mutable
 trait AbstractTesterThread
 
 class TesterThreadList(protected val elts: Seq[AbstractTesterThread]) {
-  def toSeq(): Seq[AbstractTesterThread] = elts
+  def toSeq: Seq[AbstractTesterThread] = elts
 
-  def join() {
+  def join: Unit = {
     Context().backend.doJoin(elts, None)
   }
```
Comment (jackkoenig) [inline src/main/scala/chiseltest/internal/BackendInterface.scala:18]: I actually think `join` should keep the `()`. The idea is that if a 0-arity function has side-effects, it should have the `()`. `Iterator.hasNext` vs. `Iterator.next()` is a good example: https://www.scala-lang.org/api/2.13.8/scala/collection/Iterator.html

Comment (ekiwi) [reply src/main/scala/chiseltest/internal/BackendInterface.scala:18]: I think you are right. I will change it.

Comment (ekiwi) [reply src/main/scala/chiseltest/internal/BackendInterface.scala]: changed

**Review by jackkoenig — APPROVED** [review]

**Pull request merged**